### PR TITLE
feat: migrate less to token for Segmented

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- https://github.com/ant-design/ant-design/issues/41884

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | migrate less to token for Segmented  |
| 🇨🇳 Chinese | 迁移 `Segmented` 组件的 `less token`  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 01eaeae</samp>

This pull request introduces a new `Segmented` component that uses component tokens to customize its style. It also refactors the component token interface and updates the documentation for less variables and component tokens in both English and Chinese.

### <samp>🤖 Generated by Copilot at 01eaeae</samp>

*  Move `ComponentToken` interface to top level and define common properties for segmented component ([link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-d8755e0c6400f3ca89c70784447555131c3adf13dab1289bbd756666784ac06dL6-R12))
*  Modify `genComponentStyleHook` function to accept a function that returns the component token based on the global token ([link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-d8755e0c6400f3ca89c70784447555131c3adf13dab1289bbd756666784ac06dL196-R219))
*  Add `Segmented` section and table of less variables and component tokens for the new segmented component in both English and Chinese documentation (`docs/react/migrate-less-variables.en-US.md` and `docs/react/migrate-less-variables.zh-CN.md`) ([link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL148-R154), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L148-R154))
*  Remove unnecessary empty lines under various sections in both English and Chinese documentation (`docs/react/migrate-less-variables.en-US.md` and `docs/react/migrate-less-variables.zh-CN.md`) ([link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL42), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL56), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL137), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL157), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL167), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L42), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L56), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L137), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L157), [link](https://github.com/ant-design/ant-design/pull/42136/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L167))

### CODE REVIEW

File: components/segmented/style/index.tsx [hunk 1]
```diff
@@ -3,12 +3,13 @@ import type { FullToken, GenerateStyle } from '../../theme/internal';
...
 import { resetComponent, textEllipsis } from '../../style';
 
-export interface ComponentToken {}
-
-interface SegmentedToken extends FullToken<'Segmented'> {
+export interface ComponentToken {
   segmentedPaddingHorizontal: number;
   segmentedPaddingHorizontalSM: number;
   segmentedContainerPadding: number;
+}
```
Comment (MadCcc) [inline components/segmented/style/index.tsx:10]: 这里和表格对不上

Comment (kiner-tang) [reply components/segmented/style/index.tsx]: done

File: components/segmented/style/index.tsx [hunk 2]
```diff
@@ -193,26 +194,33 @@ const genSegmentedStyle: GenerateStyle<SegmentedToken> = (token: SegmentedToken)
...
-  } = token;
-
-  const segmentedToken = mergeToken<SegmentedToken>(token, {
-    segmentedPaddingHorizontal: token.controlPaddingHorizontal - lineWidth,
-    segmentedPaddingHorizontalSM: token.controlPaddingHorizontalSM - lineWidth,
-    segmentedContainerPadding: lineWidthBold,
-    labelColor: colorTextLabel,
-    labelColorHover: colorText,
-    bgColor: colorBgLayout,
-    bgColorHover: colorFillSecondary,
```
Comment (afc163) [inline components/segmented/style/index.tsx:214]: 既有 bgColor 也有 colorBgXxxx，命名顺序感觉不太对。

File: components/segmented/demo/componentToken.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { ConfigProvider, Segmented } from 'antd';
+
+export default () => (
+  <ConfigProvider
+    theme={{
+      components: {
+        Segmented: {
+          segmentedContainerPadding: 50,
+          labelColor: '#222',
```
[the remark this answers is not in the document]
Comment (afc163) [reply components/segmented/demo/componentToken.tsx:10]: 和这个 buttonFontWeight 是一类问题：https://github.com/ant-design/ant-design/pull/42121#discussion_r1185653615

[the remark this answers is not in the document]
Comment (kiner-tang) [reply components/segmented/demo/componentToken.tsx:10]: 之前的变量命名就是这样，此次没改名字，只是把一些 token 抽离到 componentToken

File: components/segmented/style/index.tsx [hunk 4]
```diff
@@ -3,12 +3,8 @@ import { resetComponent, textEllipsis } from '../../style';
...
 import { genComponentStyleHook, mergeToken } from '../../theme/internal';
 
-export interface ComponentToken {}
-
-interface SegmentedToken extends FullToken<'Segmented'> {
-  segmentedPaddingHorizontal: number;
-  segmentedPaddingHorizontalSM: number;
-  segmentedContainerPadding: number;
+export interface ComponentToken {
+  padding: number;
```
Comment (MadCcc) [inline components/segmented/style/index.tsx:7]: 这个就不用加了，一是 v4 没有，二是和 global token 重名了

File: docs/react/migrate-less-variables.zh-CN.md [hunk 5]
```diff
@@ -150,7 +150,17 @@ title: Less 变量迁移 Design Token
...
+
+<!-- prettier-ignore -->
+| Less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@segmented-container-padding` | `padding` | - |
+| `@segmented-label-color` | `labelColor` | - |
+| `@segmented-bg` | `bgColor` | - |
+| `@segmented-hover-bg` | `bgColorHover` | - |
+| `@segmented-label-hover-color` | `labelColorHover` | - |
+| `@segmented-selected-bg` | `bgColorSelected` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:163]:
```suggestion
| `@segmented-label-color` | `itemColor` | - |
| `@segmented-bg` | `bgColor` | - |
| `@segmented-hover-bg` | `itemHoverBg` | - |
| `@segmented-label-hover-color` | `itemHoverColor` | - |
| `@segmented-selected-bg` | `itemSelectedBg` | - |
```

File: components/segmented/style/index.tsx (hunk 4 above)
Comment (kiner-tang) [reply components/segmented/style/index.tsx:7]: 好

File: components/segmented/style/index.tsx [hunk 6]
```diff
@@ -3,18 +3,19 @@ import { resetComponent, textEllipsis } from '../../style';
 import type { FullToken, GenerateStyle } from '../../theme/internal';
 import { genComponentStyleHook, mergeToken } from '../../theme/internal';
 
-export interface ComponentToken {}
+export interface ComponentToken {
+  itemColor: string;
+  itemHoverColor: string;
+  bgColor: string;
```
Comment (MadCcc) [inline components/segmented/style/index.tsx:9]: 这个不透出了吧，可以直接用 style 覆盖的

**Review by MadCcc — APPROVED** [review]

**Pull request merged**